functions.py

```diff
@@ -0,0 +1,262 @@
+
+    if pd.notnull(votes):
+        facteur_2 = gaussian_filter(votes, max_users, sigma)
+    else:
+        facteur_2 = 0
+        
+    if sequel(title_main, titre):
+        note = 0
+    else:
+        note = imdb_score**2 * facteur_1 * facteur_2
+    
+    return note
+
+# Complete the film_selection list which contains 5 films that will be recommended to the user. The films are selected from the parametres_films list and are taken into account only if the title is different enough from other film titles.
+def add_to_selection(film_selection, parametres_films):    
+    film_list = film_selection[:]
+    icount = len(film_list)    
+    for i in range(31):
+        already_in_list = False
+        for s in film_selection:
+            if s[0] == parametres_films[i][0]: already_in_list = True
+            if sequel(parametres_films[i][0], s[0]): already_in_list = True            
+        if already_in_list: continue
+        icount += 1
+        if icount <= 3:
+            film_list.append(parametres_films[i])
+    return film_list
+
+# Removes sequels from the list if more than two films from a series are present. The older one is kept.
+def remove_sequels(film_selection):    
+    removed_from_selection = []
+    for i, film_1 in enumerate(film_selection):
+        for j, film_2 in enumerate(film_selection):
+            if j <= i: continue 
+            if sequel(film_1[0], film_2[0]): 
+                last_film = film_2[0] if film_1[1] < film_2[1] else film_1[0]
+                removed_from_selection.append(last_film)
+
+    film_list = [film for film in film_selection if film[0] not in removed_from_selection]
+
+    return film_list   
+
+# Creates a list of 5 films that will be recommended to the user.
+def find_similarities(df, id_entry, del_sequels = True, verbose = False):    
+    if verbose: 
+        print(90*'_' + '\n' + "QUERY: films similar to id={} -> '{}'".format(id_entry,
+                                df.iloc[id_entry]['title']))
+    #____________________________________
+    liste_films = recommand(df, id_entry)
+    #__________________________________
+    # Create a list of 31 films
+    parametres_films = extract_parameters(df, liste_films)
+    #_______________________________________
+    # Select 5 films from this list
+    film_selection = []
+    film_selection = add_to_selection(film_selection, parametres_films)
+    #__________________________________
+    # delation of the sequels
+    if del_sequels: film_selection = remove_sequels(film_selection)
+    #______________________________________________
+    # add new films to complete the list
+    film_selection = add_to_selection(film_selection, parametres_films)
+    #_____________________________________________
+    selection_titres = []
+    for i,s in enumerate(film_selection):
+        selection_titres.append([s[0].replace(u'\xa0', u''), s[3]])
+        if verbose: print("nº{:<2}     -> {:<30}".format(i+1, s[0]))
+
+    return selection_titres
```

Comment:
Like Phil mentioned, this was done through peer programming. It worked like a charm but looks like we forgot an EOL.